src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayCompilerInternalOptions.cs

```diff
@@ -52,6 +52,8 @@ internal enum SymbolDisplayCompilerInternalOptions
 
         /// <summary>
         /// Append '!' to non-nullable reference types.
+        /// Note this causes SymbolDisplay to pull on IsNullable and therefore NonNullTypes,
+        /// so don't use this option in binding, in order to avoid cycles.
```

Comment:
Can we add an assert in the binder to verify this?

Reply:
I'm not sure how we could prevent this flag from being used in binder code.